### PR TITLE
feat: add max margin and LP return

### PR DESCRIPTION
**Pull request by Jiajia-Cui**

### CODE REVIEW

File: vega_sim/parameter_test/parameter/loggers.py [hunk 1]
```diff
@@ -126,17 +133,38 @@ def _ideal_market_maker_single_data_extraction(
...
 
     additional_fns = additional_data_fns if additional_data_fns is not None else []
+
+    if len(state_values) == 0:
+        max_margin = margin_lp
+    else:
+        old_logs = state_values[mm_agent.current_step-2]
+        max_margin = old_logs["LP: Max Margin"]
+        max_margin = max(margin_lp, max_margin)
+
```
Comment (Galen-Cao) [inline vega_sim/parameter_test/parameter/loggers.py:151]: Could we maybe put calculation of max_margin & max_lockedCapital together to simplify the code? And maybe let `old_logs = state_values[-1]` should have the same result.

**Comment by TomMcL** [comment]
What are the changes in the DemoNotebook here?

File: vega_sim/parameter_test/parameter/loggers.py (hunk 1 above)
Comment (TomMcL) [reply vega_sim/parameter_test/parameter/loggers.py:151]: Agreed on both

File: vega_sim/parameter_test/parameter/loggers.py [hunk 2]
```diff
@@ -126,17 +133,38 @@ def _ideal_market_maker_single_data_extraction(
...
+    if len(state_values) == 0:
+        max_lockedCapital = margin_lp + bond_lp
+    else:
+        old_logs = state_values[mm_agent.current_step-2]
+        max_lockedCapital = old_logs["LP: Max Locked Capital"]
+        max_lockedCapital = max(margin_lp + bond_lp, max_lockedCapital)
```
Comment (TomMcL) [inline vega_sim/parameter_test/parameter/loggers.py]: `max_locked_capital`

**Comment by Jiajia-Cui** [comment]
> 

added the new outputs in `experiment.plot_results`

File: vega_sim/parameter_test/parameter/loggers.py [hunk 3]
```diff
@@ -62,6 +66,7 @@ def ideal_market_maker_single_data_extraction(
 def _ideal_market_maker_single_data_extraction(
     vega: VegaServiceNull,
     agents: List[Agent],
+    state_values: List = None,
```
Comment (TomMcL) [inline vega_sim/parameter_test/parameter/loggers.py:73]: Still not really sure I like this vs just calculating derived values at a later time, though in the interests of expediency we can probably stick with it for now

**Review by TomMcL — CHANGES_REQUESTED** [review]
Needs a `black` run too :+1:

**Comment by TomMcL** [comment]
Needs another `make black` please 😄 (You can enable in VSCode so that it runs on save, which makes things easier!) 

Also, the DemoNotebook.ipynb file isn't for actual investigation, should just be a useful template to copy into the specific folders for each investigation. This one still looks reasonable though so will leave it for now!

**Review by TomMcL — APPROVED** [review]

**Pull request merged**